components/src/core/components/Input/Time/TimeInput.vue

```diff
@@ -110,23 +133,28 @@ export default defineComponent({
       type: Number,
       default: 1,
     },
+    is24HrsFormat: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
   },
 
   setup(props, context) {
     const timePickerOpen = ref<boolean>(false);
     const amPmLabelFocus = ref<boolean>(false);
     let inputTime = '';
 
-    interface TimeInputState {
-      time: string;
-      am: boolean;
-      pickerInput: string | null;
-    }
+    const timeFormatToParse = props.is24HrsFormat
+      ? TIME_FORMAT_24_HR
+      : TIME_FORMAT_12_HR_WITH_PERIOD;
+    const timeFormatToDisplay = props.is24HrsFormat
+      ? TIME_FORMAT_24_HR
+      : TIME_FORMAT_12_HR;
 
     const state = reactive<TimeInputState>({
       time: '',
-      am: true,
       pickerInput: props.modelValue,
+      ...(!props.is24HrsFormat && {am: true}),
```

Comment:
am will be added to state here if only your prop is there. this will break the watcher?

```
const state = reactive({
   am: !!prop.is24HrsFormat
})
```